drivers/display/ssd1306.c

```diff
@@ -373,23 +374,47 @@ static int ssd1306_set_orientation(const struct device *dev,
 static int ssd1306_set_pixel_format(const struct device *dev,
 				    const enum display_pixel_format pf)
 {
-	if (pf == PIXEL_FORMAT_MONO10) {
+	struct ssd1306_data *data = dev->data;
+
+	uint8_t cmd;
+	int ret;
+
+	if (pf == data->pf) {
 		return 0;
 	}
-	LOG_ERR("Unsupported");
-	return -ENOTSUP;
+
+	if (pf == PIXEL_FORMAT_MONO10) {
```

Comment:
Am I missing something here? Shouldn't the conditional be something like:

```
if (pf == PIXEL_FORMAT_MONO10) {
    cmd = SSD1306_SET_NORMAL_DISPLAY;
} else if (pf == PIXEL_FORMAT_MONO01) {
   cmd == SSD1306_SET_REVERSE_DISPLAY;
} else {
   LOG_ERR("Unsupported");
   return -ENOTSUP;
}
```

Reply:
good catch, of course it should. Thank you!